fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/ChainedCursor.java

```diff
@@ -53,33 +107,111 @@
+
     // for detecting incorrect cursor usage
     private boolean mayGetContinuation = false;
 
+    /**
+     * Creates a new <code>ChainedCursor</code>.  When created in this fashion any resource limits that may be
+     * specified in cursors that are used within the <code>nextGenerator</code> will be honored.
+     *
+     * @param nextGenerator a function that that is called to perform iteration
+     * @param continuationEncoder a function that takes a value that was returned by <code>nextGenerator</code> and
+     *     encodes it as a continuation value
+     * @param continuationDecoder a function that takes a continuation that was produced by <code>continuationEncoder</code>
+     *     and decodes it back to its original value as returned by <code>nextGenerator</code>
+     * @param continuation the initial continuation for the iteration
+     * @param executor executor that will be returned by {@link #getExecutor()}
+     */
+    public ChainedCursor(
+            @Nonnull Function<Optional<T>, CompletableFuture<Optional<T>>> nextGenerator,
+            @Nonnull Function<T, byte[]> continuationEncoder,
+            @Nonnull Function<byte[], T> continuationDecoder,
+            @Nullable byte[] continuation,
+            @Nonnull Executor executor) {
+        this(null, nextGenerator, continuationEncoder, continuationDecoder, continuation, null, executor);
+    }
+
+    /**
+     * Creates a <code>ChainedCursor</code>.
+     *
+     * <p>The {@link ScanProperties#isReverse()}} cannot be honored by the <code>ChainedCursor</code> as the
+     * direction of iteration is strictly controlled by the <code>nextGenerator</code>.
+     *
+     * @param context a record context
+     * @param nextGenerator a function that that is called to perform iteration
+     * @param continuationEncoder a function that takes a value that was returned by <code>nextGenerator</code> and
+     *     encodes it as a continuation value
+     * @param continuationDecoder a function that takes a continuation that was produced by <code>continuationEncoder</code>
+     *     and decodes it back to its original value as returned by <code>nextGenerator</code>
+     * @param continuation the initial continuation for the iteration
+     * @param scanProperties properties used to control the scanning behavior
+     */
     public ChainedCursor(
+            @Nonnull FDBRecordContext context,
             @Nonnull Function<Optional<T>, CompletableFuture<Optional<T>>> nextGenerator,
-            @Nonnull Function<T, byte[]> getContinuation,
-            @Nonnull Function<byte[], T> getContinueValue,
+            @Nonnull Function<T, byte[]> continuationEncoder,
+            @Nonnull Function<byte[], T> continuationDecoder,
             @Nullable byte[] continuation,
+            @Nonnull ScanProperties scanProperties) {
+        this(context, nextGenerator, continuationEncoder, continuationDecoder, continuation, scanProperties, context.getExecutor());
```

Comment:
Since we can't honour the `ScanProperties.isReverse()`, perhaps we should check for it here and throw and exception (or at least log) if `ScanProperties.isReverse()` is `true`?